### PR TITLE
Add plugin top-level configuration

### PR DESCRIPTION
**Pull request by teresam856**
Proposal for [issue 1535](https://github.com/GMOD/jbrowse-components/issues/1535)

Adds method to PluginManager that retrieves plugin's configuration objects if they exist
Modifies the Jbrowse Model to add the plugin's configurations 
Adds optional configuration param to the base class for a JBrowse plugin 
Adds a test and a test plugin to verify we are able to read and get from the configuration

### CODE REVIEW

File: products/jbrowse-web/src/tests/JBrowse.test.js [hunk 1]
```diff
@@ -65,6 +70,29 @@ test('lollipop track test', async () => {
...
+  })
+  rootModel.setDefaultSession()
+  pluginManager.setRootModel(rootModel)
+  pluginManager.configure()
+  const state = pluginManager.rootModel
+  const { jbrowse } = state
+  const { configuration } = jbrowse
+  // test reading top level configurations added by Test Plugin
+  const test = getConf(jbrowse, ['TestPlugin', 'topLevelTest'])
+  const test2 = readConfObject(configuration, ['TestPlugin', 'topLevelTest'])
```
Comment (teresam856) [inline products/jbrowse-web/src/tests/JBrowse.test.js:91]: Note: will need to be change once jexl get's merged.

File: packages/core/Plugin.ts [hunk 2]
```diff
@@ -9,6 +10,8 @@ export default abstract class Plugin {
   install(_pluginManager: PluginManager): void {}
 
   configure(_pluginManager: PluginManager): void {}
+
+  configuration: AnyConfigurationSchemaType | undefined = undefined
```
Comment (rbuels) [inline packages/core/Plugin.ts:14]: probably want to rename this to `configurationSchema`, cause it is a schema, not a config with data in it

File: products/jbrowse-web/src/jbrowseModel.js [hunk 3]
```diff
@@ -39,6 +39,7 @@ export default function JBrowseWeb(
           defaultValue: false,
         },
         theme: { type: 'frozen', defaultValue: {} },
+        ...pluginManager.pluginConfigurationSchemas(),
```
Comment (rbuels) [inline products/jbrowse-web/src/jbrowseModel.js:42]: So this has the effect of having the top-level configuration have keys like:

```
MyPluginName: { bunch of config for MyPlugin }
```

right?

**Review by rbuels — COMMENTED** [review]
this is very elegant :+1:

just one little rename needed I think

File: products/jbrowse-web/src/jbrowseModel.js (hunk 3 above)
Comment (rbuels) [reply products/jbrowse-web/src/jbrowseModel.js:42]: yeah, seems like it, based on the tests

File: packages/core/Plugin.ts (hunk 2 above)
Comment (teresam856) [reply packages/core/Plugin.ts:14]: You got it! 👍  Will go ahead and make that change.

**Comment by rbuels** [comment]
looks great!

**Pull request merged**